File: back-end/reservasti/src/main/java/br/com/reservasti/domain/helpdesk/Chamado.java
package br.com.reservasti.domain.helpdesk;


import br.com.reservasti.domain.equipamento.Equipamento;
import br.com.reservasti.domain.funcionario.Funcionario;
import br.com.reservasti.domain.helpdesk.dto.AberturaChamadoDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "chamados")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Chamado {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    private Long versao;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "equipamento_id")
    private Equipamento equipamento;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "solicitante_id")
    private Funcionario solicitante;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tecnico_id")
    private Funcionario tecnico;

    @Enumerated(EnumType.STRING)
    private TipoProblema tipoProblema;

    @Column(columnDefinition = "TEXT")
    private String descricaoDetalhada;

    @Column(name = "localizacao_exata")
    private String localizacao;

    @Enumerated(EnumType.STRING)
    private StatusChamado status;

    private LocalDateTime dataAbertura;
    private LocalDateTime dataResolucao;

    public Chamado(AberturaChamadoDTO dto,Funcionario solicitante){
        this.solicitante=solicitante;
        this.tipoProblema=dto.tipoProblema();
        this.descricaoDetalhada= dto.descricaoDetalhada();
        this.localizacao=dto.localizacao();
        this.dataAbertura=LocalDateTime.now();
    }
}

File: front-end/reservasti/src/app/core/models/chamado.ts

export interface Page<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

export interface ResumoChamadoDTO {
  id: number;
  nomeSolicitante: string;
  tipoProblema: string;
  localizacao: string;
  status: string;
  descricao: string;
  dataAbertura: string;
}

export interface DetalhamentoChamadoDTO {
  id: number;
  nomeSolicitante: string;
  nomeTecnico: string;
  tipoProblema: string;
  status: string;
  dataAbertura: string;
  posicaoFila?: number;
}
export interface AberturaChamadoDTO {
  equipamentoId?: number | null;
  solicitanteId: number;
  tipoProblema: string;
  descricaoDetalhada: string;
  localizacao: string;
  tecnicoId?: number | null;
}
